pkg/restore/restore.go

```diff
@@ -917,44 +930,60 @@ func (r *pvRestorer) executePVAction(obj *unstructured.Unstructured) (*unstructu
 		return obj, nil
 	}
 
-	// If we can't find a snapshot record for this particular PV, it most likely wasn't a PV that Ark
-	// could snapshot, so return early instead of trying to restore from a snapshot.
-	backupInfo, found := r.volumeBackups[pvName]
-	if !found {
-		return obj, nil
-	}
+	// TODO: Determine if we check for existence of old backup format here
 
-	// Past this point, we expect to be doing a restore
+	for _, snapshot := range r.volumeSnapshots {
+		location, err := r.snapshotLocationLister.VolumeSnapshotLocations(r.backupNamespace).Get(snapshot.Spec.Location)
+		if err != nil {
+			return nil, err
+		}
+		blockStore, err := r.blockStoreGetter.GetBlockStore(location.Spec.Provider)
+		if err != nil {
+			return nil, err
+		}
 
-	if r.blockStore == nil {
-		return nil, errors.New("you must configure a persistentVolumeProvider to restore PersistentVolumes from snapshots")
-	}
+		// Should we check for nil bs here?
+		// An old backup that refers to a bs that isn't configured anymore?
 
-	log := r.logger.WithFields(
-		logrus.Fields{
-			"persistentVolume": pvName,
-			"snapshot":         backupInfo.SnapshotID,
-		},
-	)
+		if err := blockStore.Init(location.Spec.Config); err != nil {
+			return nil, err
+		}
 
-	log.Info("restoring persistent volume from snapshot")
-	volumeID, err := r.blockStore.CreateVolumeFromSnapshot(backupInfo.SnapshotID, backupInfo.Type, backupInfo.AvailabilityZone, backupInfo.Iops)
-	if err != nil {
-		return nil, err
-	}
-	log.Info("successfully restored persistent volume from snapshot")
+		if _, err := blockStore.GetVolumeID(obj); err != nil {
```

Comment:
we should still check this once we've gotten the snapshot & VSL to make sure it returns a value, but if it doesn't, then we should return an error rather than continue, since there's only one possible snapshot we should be restoring.